pkgs/applications/misc/udiskie/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv, fetchFromGitHub, asciidoc-full, gettext
+, gobjectIntrospection, gtk3, hicolor_icon_theme, libnotify
+, pythonPackages, udisks2, wrapGAppsHook }:
+
+pythonPackages.buildPythonApplication rec {
+  name = "udiskie-${version}";
+  version = "1.5.1";
+
+  src = fetchFromGitHub {
+    owner = "coldfix";
+    repo = "udiskie";
+    rev = version;
+    sha256 = "01x5fvllb262x6r3547l23z7p6hr7ddz034bkhmj2cqmf83sxwxd";
+  };
+
+  preConfigure = ''
+    export XDG_RUNTIME_DIR=/tmp
```

Comment:
> you can set environment variables outside of the phases. Just passing XDG_RUNTIME_DIR=/tmp; is sufficient.

I know this feature is used all over the place, but I always felt that was weird/wrong. Like a leaky abstraction.